src/components/GridComponent.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import {useState,useEffect} from 'react'
+
+export const GridComponent = ({grid}) => {
+
+    // let products=[]
+    // grid.forEach((product,index) => {
+    //     products.push(
+    //         <div className='productContainer'>
+    //             <img key={'img'+index}  src={product.url} alt=''></img>
+    //             <p key={'name'+index}> {product.name}</p>
+    //             <p key={'price'+index}> {product.price}</p>
+    //             <p key={'category'+index}> {product.category}</p>
+    //         </div>
+                
+        
+            
+    //         // <div key={index} className='gallery_img' style={{backgroundImage:`url(${product.url})`}}></div>
+    //     )
+    // });
+    const [count,setCount]=useState(0);
+    useEffect(()=>{
+        const timerId=setInterval(()=>{
+            setCount(count=>count+1);
+        },5000)
+        return ()=>clearInterval(timerId);
+    },[]);
+
+    const tmp= <div className='productContainer'>
```

Comment:
Here you can have a smaller component <Product>, so it will keep your code smaller, cleaner and more testeable